src/hooks/datadog/metrics/webpack.ts

```diff
@@ -0,0 +1,279 @@
+
+export const getChunksFromModule = (
+    stats: StatsJson,
+    chunksPerId: { [key: string]: Chunk },
+    module: Module
+) => {
+    if (module.chunks.length) {
+        return module.chunks.map((c) => chunksPerId[c]);
+    }
+
+    // Find the chunks from the chunk list directly.
+    // Webpack may not have registered module's chunks in some cases.
+    // eslint-disable-next-line no-underscore-dangle
+    return stats.chunks.filter((c) => foundInModules(c, module.identifier || module._identifier));
+};
+
+const getMetricsFromModule = (
+    stats: StatsJson,
+    indexed: WebpackIndexedObject,
+    context: string,
+    module: Module
+) => {
+    const chunks = getChunksFromModule(stats, indexed.chunksPerId, module);
+    const entries: Set<string> = new Set();
+    for (const chunk of chunks) {
+        getEntriesFromChunk(chunk, indexed, entries);
+    }
+    const chunkTags = getChunkTags(chunks);
+    const entryTags = getEntryTags(entries);
+    const moduleName = getDisplayName(module.name, context);
+
+    return [
+        {
+            metric: 'modules.size',
+            type: 'size',
+            value: module.size,
+            tags: [
+                `moduleName:${moduleName}`,
+                `moduleType:${getType(moduleName)}`,
+                ...entryTags,
+                ...chunkTags,
+            ],
+        },
+    ];
+};
+
+export const getModules = (
+    stats: StatsJson,
+    indexed: WebpackIndexedObject,
+    context: string
+): Metric[] => {
+    return flattened(
+        Object.values(indexed.modulesPerName).map((module) => {
+            return getMetricsFromModule(stats, indexed, context, module);
+        })
+    );
+};
+
+// Find in entries.chunks
+export const getChunks = (stats: StatsJson, indexed: WebpackIndexedObject): Metric[] => {
+    const chunks = stats.chunks;
+
+    return flattened(
+        chunks.map((chunk) => {
+            const entryTags = getEntryTags(getEntriesFromChunk(chunk, indexed));
+            const chunkName = chunk.names.length ? chunk.names.join(' ') : chunk.id;
+
+            return [
+                {
+                    metric: 'chunks.size',
+                    type: 'size',
+                    value: chunk.size,
+                    tags: [`chunkName:${chunkName}`, ...entryTags],
+                },
+                {
+                    metric: 'chunks.modules.count',
+                    type: 'count',
+                    value: chunk.modules.length,
+                    tags: [`chunkName:${chunkName}`, ...entryTags],
+                },
+            ];
+        })
+    );
+};
+
+export const getAssets = (stats: StatsJson, indexed: WebpackIndexedObject): Metric[] => {
+    const assets = stats.assets;
+    return assets.map((asset) => {
+        const chunks = asset.chunks.map((c) => indexed.chunksPerId[c]);
+        const entries: Set<string> = new Set();
+        for (const chunk of chunks) {
+            getEntriesFromChunk(chunk, indexed, entries);
```

Comment:
nit: it's a little surprising to have a `getSomething` function with exclusively side effects

Reply:
Is it about the `entries` parameter?
I think it can be removed, as it's filled in by default.